### PR TITLE
 lib/imagery: fix SONAME missing

### LINKED ISSUE #2268: libgrass_imagery: missing SONAME
bisecting revealed since introduction of this line https://github.com/OSGeo/grass/blob/98dba90ea38a188559195e9d19034b47fdfefc80/lib/imagery/Makefile#L10 by 0e109f649005d10190be8e8d25012ded157e0d6c the library `libgrass_imagery.8.0.so` has no SONAME anymore.

i.e. `objdump -p ./dist.x86_64-pc-linux-gnu/lib/libgrass_imagery.8.0.so | grep SONAME` should not be empty but something like `SONAME               libgrass_imagery.8.0.so`

this missing SONAME leads to 
- downstream issues like https://github.com/qgis/QGIS/pull/47576 and 
- workarounds https://src.fedoraproject.org/rpms/qgis/c/115e2008ceee0ebcba88dcbe8580eca514f6f08d?branch=rawhide

thanks for reworking this Makefile :smile:

### PR DESCRIPTION
**Pull request by t0b3**
Fixes #2268

### CODE REVIEW

**Comment by t0b3** [comment]
shall I remove it?

**Comment by nilason** [comment]
Please give it a try locally first with the line removed. If there are no problem, update this PR to see what the CI checks will tell us.

**Comment by t0b3** [comment]
k, it compiles and the library is fixed. :smile:

File: lib/imagery/Makefile [hunk 1]
```diff
@@ -5,9 +5,9 @@ LIB = IMAGERY
 EXTRA_INC = $(VECT_INC)
 EXTRA_CFLAGS = $(VECT_CFLAGS)
 
+include $(MODULE_TOPDIR)/include/Make/Grass.make
```
Comment (nilason) [inline lib/imagery/Makefile:8]:
```suggestion
```

**Comment by nilason** [comment]
@neteler
If I understand correctly the line:
https://github.com/OSGeo/grass/blob/98dba90ea38a188559195e9d19034b47fdfefc80/lib/imagery/Makefile#L10
was added with #2091 for enabling copying of images to the documentation.
Testing locally with that line removed, suggests that it isn't needed. Could you please test this (with no import of Grass.make)?

**Comment by t0b3** [comment]
@nilason @neteler omitting Grass.make results in colortables dir not being packaged
to make those files available it would be necessary to include Grass.make
however the current (unpatched) situation isn't devlivering those files neither which might be unintentionally? :confused: 

see comparision of resulting files 

```
--- grass-8.0.1_grass_make_at_1st_rank.files 2022-03-19 11:48:46.865748161 +0100
+++ grass-8.0.1_dropped_grass_make.files     2022-03-19 11:48:55.155748044 +0100
@@ -430,63 +430,6 @@
 ./usr/lib64/grass80/docs/html/barscales/tail_checker.png
 ./usr/lib64/grass80/docs/html/barscales/up_ticks.png
 ./usr/lib64/grass80/docs/html/class_graphical.html
-./usr/lib64/grass80/docs/html/colortables/
-./usr/lib64/grass80/docs/html/colortables/aspectcolr.png
-./usr/lib64/grass80/docs/html/colortables/aspect.png
-./usr/lib64/grass80/docs/html/colortables/bcyr.png
-./usr/lib64/grass80/docs/html/colortables/bgyr.png
-./usr/lib64/grass80/docs/html/colortables/blues.png
-./usr/lib64/grass80/docs/html/colortables/byg.png
-./usr/lib64/grass80/docs/html/colortables/byr.png
-./usr/lib64/grass80/docs/html/colortables/celsius.png
-./usr/lib64/grass80/docs/html/colortables/corine.png
-./usr/lib64/grass80/docs/html/colortables/curvature.png
-./usr/lib64/grass80/docs/html/colortables/differences.png
-./usr/lib64/grass80/docs/html/colortables/elevation.png
-./usr/lib64/grass80/docs/html/colortables/etopo2.png
-./usr/lib64/grass80/docs/html/colortables/evi.png
-./usr/lib64/grass80/docs/html/colortables/fahrenheit.png
-./usr/lib64/grass80/docs/html/colortables/forest_cover.png
-./usr/lib64/grass80/docs/html/colortables/gdd.png
-./usr/lib64/grass80/docs/html/colortables/grass.png
-./usr/lib64/grass80/docs/html/colortables/greens.png
-./usr/lib64/grass80/docs/html/colortables/grey1.0.png
-./usr/lib64/grass80/docs/html/colortables/grey255.png
-./usr/lib64/grass80/docs/html/colortables/grey.eq.png
-./usr/lib64/grass80/docs/html/colortables/grey.log.png
-./usr/lib64/grass80/docs/html/colortables/grey.png
-./usr/lib64/grass80/docs/html/colortables/gyr.png
-./usr/lib64/grass80/docs/html/colortables/haxby.png
-./usr/lib64/grass80/docs/html/colortables/inferno.png
-./usr/lib64/grass80/docs/html/colortables/kelvin.png
-./usr/lib64/grass80/docs/html/colortables/magma.png
-./usr/lib64/grass80/docs/html/colortables/ndvi.png
-./usr/lib64/grass80/docs/html/colortables/ndwi.png
-./usr/lib64/grass80/docs/html/colortables/nlcd.png
-./usr/lib64/grass80/docs/html/colortables/oranges.png
-./usr/lib64/grass80/docs/html/colortables/plasma.png
-./usr/lib64/grass80/docs/html/colortables/population_dens.png
-./usr/lib64/grass80/docs/html/colortables/population.png
-./usr/lib64/grass80/docs/html/colortables/precipitation_daily.png
-./usr/lib64/grass80/docs/html/colortables/precipitation_monthly.png
-./usr/lib64/grass80/docs/html/colortables/precipitation.png
-./usr/lib64/grass80/docs/html/colortables/rainbow.png
-./usr/lib64/grass80/docs/html/colortables/ramp.png
-./usr/lib64/grass80/docs/html/colortables/random.png
-./usr/lib64/grass80/docs/html/colortables/reds.png
-./usr/lib64/grass80/docs/html/colortables/roygbiv.png
-./usr/lib64/grass80/docs/html/colortables/rstcurv.png
-./usr/lib64/grass80/docs/html/colortables/ryb.png
-./usr/lib64/grass80/docs/html/colortables/ryg.png
-./usr/lib64/grass80/docs/html/colortables/sepia.png
-./usr/lib64/grass80/docs/html/colortables/slope.png
-./usr/lib64/grass80/docs/html/colortables/soilmoisture.png
-./usr/lib64/grass80/docs/html/colortables/srtm_plus.png
-./usr/lib64/grass80/docs/html/colortables/srtm.png
-./usr/lib64/grass80/docs/html/colortables/terrain.png
-./usr/lib64/grass80/docs/html/colortables/viridis.png
-./usr/lib64/grass80/docs/html/colortables/water.png
-./usr/lib64/grass80/docs/html/colortables/wave.png
 ./usr/lib64/grass80/docs/html/database_graphical.html
 ./usr/lib64/grass80/docs/html/database.html
 ./usr/lib64/grass80/docs/html/databaseintro.html
@@ -4675,63 +4618,6 @@
 ./usr/share/doc/grass-8.0.1/html/barscales/tail_checker.png
 ./usr/share/doc/grass-8.0.1/html/barscales/up_ticks.png
 ./usr/share/doc/grass-8.0.1/html/class_graphical.html
-./usr/share/doc/grass-8.0.1/html/colortables/
-./usr/share/doc/grass-8.0.1/html/colortables/aspectcolr.png
-./usr/share/doc/grass-8.0.1/html/colortables/aspect.png
-./usr/share/doc/grass-8.0.1/html/colortables/bcyr.png
-./usr/share/doc/grass-8.0.1/html/colortables/bgyr.png
-./usr/share/doc/grass-8.0.1/html/colortables/blues.png
-./usr/share/doc/grass-8.0.1/html/colortables/byg.png
-./usr/share/doc/grass-8.0.1/html/colortables/byr.png
-./usr/share/doc/grass-8.0.1/html/colortables/celsius.png
-./usr/share/doc/grass-8.0.1/html/colortables/corine.png
-./usr/share/doc/grass-8.0.1/html/colortables/curvature.png
-./usr/share/doc/grass-8.0.1/html/colortables/differences.png
-./usr/share/doc/grass-8.0.1/html/colortables/elevation.png
-./usr/share/doc/grass-8.0.1/html/colortables/etopo2.png
-./usr/share/doc/grass-8.0.1/html/colortables/evi.png
-./usr/share/doc/grass-8.0.1/html/colortables/fahrenheit.png
-./usr/share/doc/grass-8.0.1/html/colortables/forest_cover.png
-./usr/share/doc/grass-8.0.1/html/colortables/gdd.png
-./usr/share/doc/grass-8.0.1/html/colortables/grass.png
-./usr/share/doc/grass-8.0.1/html/colortables/greens.png
-./usr/share/doc/grass-8.0.1/html/colortables/grey1.0.png
-./usr/share/doc/grass-8.0.1/html/colortables/grey255.png
-./usr/share/doc/grass-8.0.1/html/colortables/grey.eq.png
-./usr/share/doc/grass-8.0.1/html/colortables/grey.log.png
-./usr/share/doc/grass-8.0.1/html/colortables/grey.png
-./usr/share/doc/grass-8.0.1/html/colortables/gyr.png
-./usr/share/doc/grass-8.0.1/html/colortables/haxby.png
-./usr/share/doc/grass-8.0.1/html/colortables/inferno.png
-./usr/share/doc/grass-8.0.1/html/colortables/kelvin.png
-./usr/share/doc/grass-8.0.1/html/colortables/magma.png
-./usr/share/doc/grass-8.0.1/html/colortables/ndvi.png
-./usr/share/doc/grass-8.0.1/html/colortables/ndwi.png
-./usr/share/doc/grass-8.0.1/html/colortables/nlcd.png
-./usr/share/doc/grass-8.0.1/html/colortables/oranges.png
-./usr/share/doc/grass-8.0.1/html/colortables/plasma.png
-./usr/share/doc/grass-8.0.1/html/colortables/population_dens.png
-./usr/share/doc/grass-8.0.1/html/colortables/population.png
-./usr/share/doc/grass-8.0.1/html/colortables/precipitation_daily.png
-./usr/share/doc/grass-8.0.1/html/colortables/precipitation_monthly.png
-./usr/share/doc/grass-8.0.1/html/colortables/precipitation.png
-./usr/share/doc/grass-8.0.1/html/colortables/rainbow.png
-./usr/share/doc/grass-8.0.1/html/colortables/ramp.png
-./usr/share/doc/grass-8.0.1/html/colortables/random.png
-./usr/share/doc/grass-8.0.1/html/colortables/reds.png
-./usr/share/doc/grass-8.0.1/html/colortables/roygbiv.png
-./usr/share/doc/grass-8.0.1/html/colortables/rstcurv.png
-./usr/share/doc/grass-8.0.1/html/colortables/ryb.png
-./usr/share/doc/grass-8.0.1/html/colortables/ryg.png
-./usr/share/doc/grass-8.0.1/html/colortables/sepia.png
-./usr/share/doc/grass-8.0.1/html/colortables/slope.png
-./usr/share/doc/grass-8.0.1/html/colortables/soilmoisture.png
-./usr/share/doc/grass-8.0.1/html/colortables/srtm_plus.png
-./usr/share/doc/grass-8.0.1/html/colortables/srtm.png
-./usr/share/doc/grass-8.0.1/html/colortables/terrain.png
-./usr/share/doc/grass-8.0.1/html/colortables/viridis.png
-./usr/share/doc/grass-8.0.1/html/colortables/water.png
-./usr/share/doc/grass-8.0.1/html/colortables/wave.png
 ./usr/share/doc/grass-8.0.1/html/database_graphical.html
 ./usr/share/doc/grass-8.0.1/html/database.html
 ./usr/share/doc/grass-8.0.1/html/databaseintro.html
```

**Comment by nilason** [comment]
> @nilason @neteler omitting Grass.make results in colortables dir not being packaged to make those files available it would be necessary to include Grass.make however the current (unpatched) situation isn't devlivering those files neither which might be unintentionally? 😕
> 

I'm not sure I understand what you mean. Eventual differences with the colortables dir in docs is unrelated to this Makefile (and the proposed changes of the same). That being said, `diff -rq` the result of compiling both with and without including Grass.make in this `grass/lib/imagery/Makefile` file showed no difference (in files present or not).

**Comment by t0b3** [comment]
@nilason your absolutely right :+1: I had a local caching issue causing that difference.

**Review by nilason — APPROVED** [review]

**Comment by nilason** [comment]
@neteler According to my testing, this is good to go. If you agree, please go ahead and merge.

**Comment by neteler** [comment]
No problems on Fedora 35, will merge and backport it.
Thanks for the PR!

**Comment by thesamesam** [comment]
@t0b3 could you backport this for gentoo?

**Comment by t0b3** [comment]
@neteler what would be the release date for 8.0.2?

**Comment by t0b3** [comment]
@thesamesam sure I can backport if there won't be a release these days. Would you prefer to hotfix and keep existing `arch` or better stay conservative and reset to `~arch`?

**Comment by neteler** [comment]
> @neteler what would be the release date for 8.0.2?

We are still stuck on a couple of backports which I try to clarify by next week. Then the RC cycle may start with the final release coming in a few weeks.

**Comment by t0b3** [comment]
Thanks for your information. Good luck with the backports then.

**Pull request merged**